docs/frontend/webpack-source-maps.md

```diff
@@ -6,8 +6,402 @@ which allows a developer to view / debug developer-friendly source code instead
+modern browsers implement **source maps** whereby minified code can be
+mapped to real developer sources before execution to give developers a much
+better debugging experience.
+
+If you are not familiar with how source maps work, please take a moment to
+review a couple introductory articles:
+
+* http://www.html5rocks.com/en/tutorials/developertools/sourcemaps/
+* https://github.com/ryanseddon/source-map/wiki/Source-maps:-languages,-tools-and-other-info
+* https://blog.sentry.io/2015/10/29/debuggable-javascript-with-source-maps.html
+
+The brief summary of how source maps work is as follows:
+
+* During bundling, an application is configured to output a source mapping file
+  corresponding to the application bundle.
+* The real application bundle is downloaded and parsed by a browser.
+* If the browser encounters a special control comment like:
+  `//# sourceMappingURL=PATH/TO/bundle.js.map` **and** the developer console
+  is open, then the source map file is downloaded from the URL in the control
+  comment.
+* Once downloaded, the mapped file is swapped in and debuggable in place of the
+  real (probably unreadable) application bundle.
+
+This functionality is useful in both development and production. In
+**development** an application may not be minified, but it is bundled into a
+single / couple of application chunks. Source maps can give developers a
+debugging experience corresponding to the _actual raw source files_ at issue.
+[SurviveJS](http://survivejs.com/webpack/developing-with-webpack/enabling-sourcemaps/)
+provides a very good tutorial in utilizing source maps for easier development
+and we won't touch on the topic further, as we're focused on the production
+story for this current document.
+
+In **production** source maps are the lifeline that allows application
+developers to not have to make sense of gobs of minified variables and code.
+The source maps are not just a convenience, they're practically essential for
+being able to debug a production app for any sizeable bundle in the modern
+web. The rest of this page will examine enabling your critical "out" for bugs
+you find in production.
+
+**Side Note: source maps for end users**: There is sometimes a concern that for
+most end users of an application that they shouldn't have the penalty of either
+downloading or 404'ing source maps in the non-development use of a production
+application. Fortunately, source map integration in browsers works such that
+a browser will not even attempt to download source maps unless the development
+console of the browser is open, using that as a good heuristic of when to add
+the enhanced debugging capabilities of source maps. Thus, for normal end users
+there is absolutely no penalty for having a source map integration strategy
+in production applications.
+
+##### Integrating Source Maps
+
+With the above background in mind, the big issue is where and how to serve
+source maps when focusing on production.
+
+###### Open Source Apps
```

Comment:
Doesn't this really pertain to any app, as long as IP protection isn't a priority? That would seem a more helpful dichotomy. Or maybe I'm out of the loop - are a ton of people really obfuscating their JS to protect IP these days? (honest question)